### ISSUE TITLE
Incorrect broadcasting of `when/then` inside `.agg()`

### ISSUE THREAD

**Issue by cmdlineluser** [pola-rs/polars#12382]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

pl.Config(fmt_table_cell_list_len=-1, fmt_str_lengths=120)

df = pl.DataFrame({"id": [1, 1], "value": [0, 3]})

(df.group_by("id")
   .agg(
      pl.when(pl.int_range(0, 5).is_in("value"))
        .then(pl.lit("yes"))
    )
)
 
# shape: (1, 2)
# ┌─────┬───────────────┐
# │ id  ┆ literal       │
# │ --- ┆ ---           │
# │ i64 ┆ list[str]     │
# ╞═════╪═══════════════╡
# │ 1   ┆ ["yes", null] │
# └─────┴───────────────┘ 
```

### Issue description

This is closely related to #12242 - it may make sense to deal with both at the same time.

Unless I'm mistaken, I was expecting the example to produce 5 values:

```python
shape: (1, 2)
┌─────┬──────────────────────────────────┐
│ id  ┆ literal                          │
│ --- ┆ ---                              │
│ i64 ┆ list[str]                        │
╞═════╪══════════════════════════════════╡
│ 1   ┆ ["yes", null, null, "yes", null] │
└─────┴──────────────────────────────────┘
```

If I add a match for the case of both Literals, I get the expected output:

https://github.com/pola-rs/polars/blob/4a58499bdf97203c940a3c82c1113905e8c6087d/crates/polars-lazy/src/physical_plan/expressions/ternary.rs#L167

```rust
+            (Literal(_), Literal(_)) => {
+                finish_as_iters(ac_truthy, ac_falsy, ac_mask)
+            },
+
```

However, that whole section of code has quite a bit going on, so it's possible I'm missing something.

### Expected behavior

```
shape: (1, 2)
┌─────┬──────────────────────────────────┐
│ id  ┆ literal                          │
│ --- ┆ ---                              │
│ i64 ┆ list[str]                        │
╞═════╪══════════════════════════════════╡
│ 1   ┆ ["yes", null, null, "yes", null] │
└─────┴──────────────────────────────────┘
```

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.13
Index type:          UInt32
Platform:            macOS-12.6.7-arm64-arm-64bit
Python:              3.11.6 (main, Oct  2 2023, 20:46:17) [Clang 14.0.0 (clang-1400.0.29.202)]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              2023.6.0
gevent:              <not installed>
matplotlib:          <not installed>
numpy:               1.25.0
openpyxl:            <not installed>
pandas:              2.0.3
pyarrow:             12.0.1
pydantic:            <not installed>
pyiceberg:           <not installed>
pyxlsb:              <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>
```

</details>

**Comment by romanovacca** [comment]
@stinodego I think this can be closed.

**Comment by cmdlineluser** [comment]
Fixed by [#12529](https://github.com/pola-rs/polars/pull/12529)
